tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -334,6 +370,50 @@ private function initializeData(array $initialData, array $expectedData = null)
         }
     }
 
+    private function prepareBulkWriteArguments($arguments)
+    {
+        $operations = [];
+        $operations['options'] = $arguments['options'];
+        foreach ($arguments['requests'] as $requests) {
+            if (isset($requests['arguments']['arrayFilters'])) {
+                if (isset($requests['arguments']['options'])) {
+                    array_push($requests['arguments']['options'], ['arrayFilters' => $requests['arguments']['arrayFilters']]);
+                } else {
+                    $options = ['arrayFilters' => $requests['arguments']['arrayFilters']];
+                }
+            }
+            $innerArray = [];
+            switch ($requests['name']) {
+                case 'deleteMany':
+                case 'deleteOne':
+                    $innerArray = [$requests['arguments']['filter']];
+                    if (isset($requests['arguments']['options'])) {
+                         array_push($innerArray, $requests['arguments']['options']);
+                    }
+                case 'insertOne':
+                    $innerArray = [$requests['arguments']['document']];
+                case 'replaceOne':
+                    $innerArray = [$requests['arguments']['filter'], $requests['arguments']['replacement']];
+                     if (isset($requests['arguments']['options'])) {
+                         array_push($innerArray, $requests['arguments']['options']);
+                    }
+                case 'updateMany':
+                case 'updateOne':
+                    if (isset($options)) {
+                        if (isset($requests['arguments']['options'])) {
+                            array_push($requests['arguments']['options'], $options);
+                        } else {
+                            $requests['arguments']['options'] = $options;
+                        }
+                    }
+                    $innerArray = [$requests['arguments']['filter'], $requests['arguments']['update'], $requests['arguments']['options']];
+           }
+            $operations[] = [$requests['name'] => $innerArray];
```

Comment:
I noticed a few lines in this method had some indentation issues. Please double-check later.